File: src/Models/StoreProduct.cs
namespace volantis_sms.Models
{
    public class StoreProduct
    {
        public int StoreId { get; set; }
        public Store Store { get; set; } = default!;
        public int ProductId { get; set; }
        public Product Product { get; set; } = default!;
        public decimal BuyPrice { get; set; }
        public decimal SellPrice { get; set; }
        public decimal Igv { get; set; }
        public decimal Profit { get; set; }
        public int Stock { get; set; }
        public bool Salable { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
    }
}

File: frontend/src/types/storeProduct.ts
import type { Product } from './product';
import type { Store } from './store';

export interface StoreProduct {
  storeId: number;
  productId: number;
  buyPrice: number;
  sellPrice: number;
  igv: number;
  profit: number;
  stock: number;
  salable: boolean;
  store?: Store;
  product?: Product;
  createdAt?: string;
  updatedAt?: string;
  deletedAt?: string | null;
}
